editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1492,7 +1492,9 @@ bool CanvasItemEditor::_gui_input_rotate(const Ref<InputEvent> &p_event) {
 			for (List<CanvasItem *>::Element *E = drag_selection.front(); E; E = E->next()) {
 				CanvasItem *canvas_item = E->get();
 				drag_to = transform.affine_inverse().xform(m->get_position());
-				canvas_item->_edit_set_rotation(snap_angle(canvas_item->_edit_get_rotation() + (drag_from - drag_rotation_center).angle_to(drag_to - drag_rotation_center), canvas_item->_edit_get_rotation()));
+				//Rotate opposite way if either global scale is inverted but local scale is not or global scale is not inverted but local scale is
+				bool opposite = (canvas_item->get_global_transform().get_scale().sign() != Vector2(1, 1)) == (canvas_item->get_transform().get_scale().sign() == Vector2(1, 1));
```

Comment:
I am not sure this condition is good. I haven't tested yet but it might not cover all use cases, especially when you have both vertical an horizontal scale flipped.

Reply:
I tested a lot of possible combinations, but probably not all of them. I'll set up a test scene that's got sets of three sprites (a child, a parent, and a grand parent) that cover all possible combinations of scale.

Reply:
If `opposite` is meant to be true only when there's an uneven number of negative coordinates in the two vectors, you can use dot product to assess it:
```
	print(Vector2(-1, 1).dot(Vector2(1, 1)))
	print(Vector2(-1, -1).dot(Vector2(1, 1)))
	print(Vector2(1, -1).dot(Vector2(1, 1)))
	print(Vector2(1, -1).dot(Vector2(1, -1)))
	print(Vector2(1, -1).dot(Vector2(-1, -1)))
	print(Vector2(1, -1).dot(Vector2(-1, 1)))
```
prints:
```
0
-2
0
2
0
-2
```
I'm a bit rusty on my math but I'm sure there must be a geometric explanation for why this would be the "proper" way to compare the signedness of two unit vectors.

Reply:
> I'm a bit rusty on my math but I'm sure there must be a geometric explanation for why this would be the "proper" way to compare the signedness of two unit vectors.

Well basically it just means that the two unit vectors are orthogonal in the Cartesian space. If they were colinear, the signs would either be the same or opposite for both coordinates.

Reply:
Yeah, I was more expecting something with a dot product too. I assume the condition might work because the signed scale can never be `Vector2(-1,-1)`, but that's kind of weird.

Reply:
I was unaware you could use dot product for these kind of situations; very clever! I updated my implementation to use dot product because I think it makes the code a bit cleaner. Everything still works as far as I am aware.